controllers/checlusterrestore/backup_data_restorer.go

```diff
@@ -425,6 +428,36 @@ func readBackupMetadata(rctx *RestoreContext, dataDir string) (*checlusterbackup
 	return backupMetadata, true, nil
 }
 
+// dropDatabases deletes Che related databased from Postgres, if any
+func dropDatabases(rctx *RestoreContext) {
+	if rctx.cheCR.Spec.Database.ExternalDb {
+		// Skip this step as there is an external server to connect to
+		return
+	}
+
+	databasesToDrop := []string{
+		rctx.cheCR.Spec.Database.ChePostgresDb,
+		"keycloak",
+	}
+
+	k8sClient := util.GetK8Client()
+	postgresPodName, err := k8sClient.GetDeploymentPod(deploy.PostgresName, rctx.namespace)
+	if err != nil {
+		// Postgres pod not found, probably it doesn't exist
+		return
+	}
+
+	for _, dbName := range databasesToDrop {
+		execReason := fmt.Sprintf("dropping %s database", dbName)
+		dropDatabaseCommand := fmt.Sprintf("psql -c \"DROP DATABASE %s;\"", dbName)
+		if output, err := k8sClient.DoExecIntoPod(rctx.namespace, postgresPodName, dropDatabaseCommand, execReason); err != nil {
```

Comment:
What about returning `err ?

Reply:
I think, we shouldn't break the process. This step is only needed when database format is changed.